clients/src/main/resources/common/message/UpdateFeaturesResponse.json

```diff
@@ -17,7 +17,7 @@
   "apiKey": 57,
   "type": "response",
   "name": "UpdateFeaturesResponse",
-  "validVersions": "0-1",
+  "validVersions": "0-2",
```

Comment:
Should we remove the `Results` field in V2?

Reply:
The way the future is structured, it expects results for all the features. https://github.com/apache/kafka/blob/be1161556423e5c30232a75a2d084cce1a5ecd57/clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java#L4555

We can do this, but that will mean I will have to change this code too.

Reply:
There is not a great way to check version here, so the only thing I can do easily is remove this line.  :( 
I'm wondering if this is the best move.

Reply:
If we make `UpdateFeaturesResponse.Results` ignorable from V2, the field will be null in V2 response. In V1 response, it can only be an empty array. We can tell whether this is V2 response based on that.

Reply:
Hmm - so we are relying on the default value being an empty list for v1? I think that will work.

Reply:
In V1, since the field is not ignorable, it can never be null.

Reply:
So the other aspect of this is that the field needs to be nullable and null by default or this won't work. I wasn't sure if we could change the default value between versions.

Reply:
Setting the field to ignorable, just means that if we set it, the serialization will remove the value. When we deserialize and try to get the value though it will still be the empty list.

Reply:
Yes, you are right about ignorable. Perhaps we can just treat topLevelError=none and an empty result array as V2 response since it's never used in V0/V1 responses.

Reply:
Ok so let me just summarize my understanding. 

V1 -- we short circuit the response when there is an error and put the error top level, but we still populate the results array for none and error cases

V2 -- we short circuit the response when there is an error and put the error top level, no results array for any case

For both the top level error will contain the feature that caused the error if there was one.

Reply:
In V1, if there is a top level error, we don't populate the results array, right?

Reply:
Right now we don't, but we also don't set a top level error if a feture fails. I can amend this to not return the results array in the error case.

Reply:
In V1, if we get a top level error like NOT_CONTROLLER, we don't populate the results array, right?

If we fail fast for V1, we could just do the same thing, i.e. set a top level error and don't populate the results array.

Reply:
Yup -- the key difference between not controller and a feature specific error is that we want the user to know which feature failed still. That's why the error message is crucial.

Reply:
Ok. I believe I've pushed the code with this implementation. Please take a look when you get a chance :)